### PR TITLE
Replay player with multiple stops

### PR DESCRIPTION
**Pull request by kmadsen**
## Description

Using the `ReplayHistoryPlayer` for waypoints is traversing a route with multiple waypoints, in the same way as the `ReplayRouteLocationEngine`. Think it's fair to move forward with using it here too. More on https://github.com/mapbox/mapbox-navigation-android/issues/2688

- [x] I have added any issue links
- [ ] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

## Screenshots or Gifs

output

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by kmadsen** [comment]
There was an odd jerk behavior near a waypoint. It goes backwards

**Comment by Guardiola31337** [comment]
Took this for a quick spin and it seems that we're using raw location updates 👀 that surfing puck 😅 

raw_location_updates_issue

**Comment by kmadsen** [comment]
> Took this for a quick spin and it seems that we're using raw location updates 👀 that surfing puck 😅

@Guardiola31337 this isn't actually changing the raw vs enhanced locations. Seems like this is changing to support freedrive? https://github.com/mapbox/mapbox-navigation-android/pull/2698/files#diff-5a4b8fa7b37f2888f5635d6c1fdc57b6L245-L258

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayRouteMapper.kt [hunk 1]
```diff
@@ -25,15 +26,48 @@ class ReplayRouteMapper(
     }
 
     /**
-     * Map a directions route into replay events.
+     * Map a [DirectionsRoute] route into replay events.
      *
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayRouteMapper.kt:30]: We need to add `startTime` to the KDoc

**Comment by kmadsen** [comment]
output

Looks like we won't go around corners sometimes

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayRouteMapper.kt [hunk 2]
```diff
@@ -25,15 +26,48 @@ class ReplayRouteMapper(
...
+            val routeLegEvents = mapToUpdateLocations(currentTime, routeLeg)
+            currentTime = routeLegEvents.lastOrNull()?.eventTimestamp ?: currentTime
+            routeEvents.addAll(routeLegEvents)
+        }
+        return routeEvents
+    }
+
+    /**
+     * Map a [RouteLeg] route into replay events.
+     *
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayRouteMapper.kt:47]: Same as above re: `startTime` `@param`

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayRouteMapper.kt [hunk 3]
```diff
@@ -25,15 +26,48 @@ class ReplayRouteMapper(
...
+            val stepEvents = mapToUpdateLocations(currentTime, stepGeometry)
+            currentTime = routeLegEvents.lastOrNull()?.eventTimestamp ?: currentTime
+            routeLegEvents.addAll(stepEvents)
+        }
+        return routeLegEvents
+    }
+
+    /**
+     * Map a geometry into replay events.
+     *
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayRouteMapper.kt:65]: Same as above re: `startTime` `@param`

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayRouteMapper.kt [hunk 4]
```diff
@@ -88,18 +122,16 @@ class ReplayRouteMapper(
      * @param location Android location to be replayed
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayRouteMapper.kt:122]: Same as above re: `startTime` `@param`

**Comment by kmadsen** [comment]
Stop navigation is working for me 🤔 

output

**Comment by kmadsen** [comment]
This isn't introducing anything new, and it is really introducing different issues. If anything, I think it will just give the ReplayHistoryPlayer a bad feeling.

Going to move forward with this https://github.com/mapbox/mapbox-navigation-android/pull/2802
It's a bigger change, but at least it fixes a lot of our replay issues

**Pull request closed without merging**